plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/metastore/glue/TestDeltaLakeConcurrentModificationGlueMetastore.java

```diff
@@ -0,0 +1,138 @@
+public class TestDeltaLakeConcurrentModificationGlueMetastore
+        extends AbstractTestQueryFramework
+{
+    private static final String CATALOG_NAME = "test_delta_lake_concurrent";
+    private static final String SCHEMA = "test_delta_lake_glue_concurrent_" + randomNameSuffix();
+    private String dataDirectory;
+    private GlueHiveMetastore metastore;
+    private AtomicBoolean failNextGlueDeleteTableCall = new AtomicBoolean(false);
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Session deltaLakeSession = testSessionBuilder()
+                .setCatalog(CATALOG_NAME)
+                .setSchema(SCHEMA)
+                .build();
+
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(deltaLakeSession).build();
+
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("data_delta_concurrent").toString();
+        HdfsConfig hdfsConfig = new HdfsConfig();
+        HdfsConfiguration hdfsConfiguration = new DynamicHdfsConfiguration(new HdfsConfigurationInitializer(hdfsConfig), ImmutableSet.of());
+        HdfsEnvironment hdfsEnvironment = new HdfsEnvironment(hdfsConfiguration, hdfsConfig, new NoHdfsAuthentication());
+        GlueMetastoreStats stats = new GlueMetastoreStats();
+        GlueHiveMetastoreConfig glueConfig = new GlueHiveMetastoreConfig()
+                .setDefaultWarehouseDir(dataDirectory);
+
+        AWSGlueAsync glueclient = createAsyncGlueClient(glueConfig, DefaultAWSCredentialsProviderChain.getInstance(), Optional.empty(), stats.newRequestMetricsCollector());
+        AWSGlueAsync proxiedGlueClient = newProxy(AWSGlueAsync.class, (proxy, method, args) -> {
+            Object result;
+            try {
+                if (method.getName().equals("deleteTable") && failNextGlueDeleteTableCall.get()) {
+                    // Simulate concurrent modifications on the table that is about to be dropped
+                    failNextGlueDeleteTableCall.set(false);
+                    throw new TrinoException(HIVE_METASTORE_ERROR, new ConcurrentModificationException("Test-simulated metastore concurrent modification exception"));
+                }
+                result = method.invoke(glueclient, args);
+            }
+            catch (InvocationTargetException e) {
+                throw e.getCause();
+            }
+            return result;
+        });
+
+        metastore = new GlueHiveMetastore(
+                hdfsEnvironment,
+                glueConfig,
+                directExecutor(),
+                new DefaultGlueColumnStatisticsProviderFactory(directExecutor(), directExecutor()),
+                proxiedGlueClient,
+                stats,
+                table -> true);
+
+        queryRunner.installPlugin(new TestingDeltaLakePlugin(Optional.of(new TestingDeltaLakeMetastoreModule(metastore)), EMPTY_MODULE));
+        queryRunner.createCatalog(CATALOG_NAME, "delta-lake");
```

Comment:
Change to `delta_lake`. https://github.com/trinodb/trino/commit/7009c8d83b650b2dac4d9bbd7da5020fb764e018